src/cli_support.ts

```diff
@@ -2,8 +2,16 @@ import * as path from 'path';
 
 // Postprocess generated JS.
 export function pathToModuleName(context: string, fileName: string): string {
+  fileName = fileName.replace(/\.js$/, '');
+
   if (fileName[0] === '.') {
+    // './foo' or '../foo'.
+    // Resolve the path against the dirname of the current module.
     fileName = path.join(path.dirname(context), fileName);
   }
-  return fileName.replace(/\.js$/, '').replace(/\//g, '.');
+  // Replace characters not supported by goog.module.
+  let moduleName =
```

Comment:
we use a different, reversible escaping scheme for goog.module internally, where we URI escape and then replace % with $. Should we use the same?